src/app/content/components/Page/PageComponent.tsx

```diff
@@ -37,6 +37,25 @@ export default class PageComponent extends Component<PagePropTypes> {
   private scrollToTopOrHashManager = stubScrollToTopOrHashManager;
   private processing: Array<Promise<void>> = [];
   private componentDidUpdateCounter = 0;
+  private usingMouseInput = true;
+
+  private handleMouseDown = () => {
+    if (!this.usingMouseInput) {
+      this.usingMouseInput = true;
+      this.highlightManager.setSnapMode?.(true);
+    }
+  };
+
+  private handleKeyDown = (event: KeyboardEvent) => {
+    const navigationKeys = [
+      'ArrowLeft', 'ArrowRight', 'ArrowUp', 'ArrowDown',
+      'Shift', 'Control', 'Meta', 'Alt', 'Tab', 'Home', 'End', 'PageUp', 'PageDown',
+    ];
+    if (this.usingMouseInput && navigationKeys.includes(event.key)) {
+      this.usingMouseInput = false;
+      this.highlightManager.setSnapMode?.(false);
+    }
+  };
```

Comment:
I think weird things will happen if you use both the mouse and keyboard at the same time, but I'm not sure I have a good solution for this so maybe it's fine.